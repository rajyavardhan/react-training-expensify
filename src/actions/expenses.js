import uuid from 'uuid';

//Action generators
//ADD_EXPENSE
export const addExpense  = ({createdAt = 0, description = '', notes = '', amount = 0} = {}) => ({
	type: 'ADD_EXPENSE',
	expense: {
		id: uuid(),
		notes,
		amount,
		description,
		createdAt
	}
});

//REMOVE EXPENSE

export const removeExpense = ({removeExpenseId = -1} = {}) => ({
	type: 'REMOVE_EXPENSE',
	removeExpenseId
});

//Edit expense
export  const editExpense = (id, updates) => ({
	type: 'EDIT_EXPENSE',
	id,
	updates
});