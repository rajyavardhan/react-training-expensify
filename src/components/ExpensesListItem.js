import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeraljs';

const ExpenseListItem = ({id, description, amount, createdAt}) => (
	<div>
		<Link to = {`/edit/${id}`}>
			<h2> {description} </h2>
		</Link>
		<p> {numeral(amount).format('$0,0.00')} 
				-
			{moment(createdAt).format('MMM Do YYYY')} 
		</p>
	</div>
);

export default ExpenseListItem;
		// <button
		// 	onClick = {() => {props.history.push('/edit/' +  props.id)}}
		// >
		// 	Edit
		// </button>