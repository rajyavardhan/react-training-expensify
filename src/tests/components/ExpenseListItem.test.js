import React from 'react';
import ExpensesListItem from '../../components/ExpensesListItem';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';

test(`Should render ExpensesListItem correctly`, () => {
	const wrapper = shallow(<ExpensesListItem {...expenses[1]} />);
	expect(wrapper).toMatchSnapshot();
});
test(`Should render one paragraph items`, () => {
	const wrapper = shallow(<ExpensesListItem {...expenses[1]}/>);
	expect(wrapper.find('p').length).toBe(1);
});
test(`Should render the span text as 109500`, () => {
	const wrapper = shallow(<ExpensesListItem {...expenses[1]}/>);
	expect(wrapper.find('h2').length).toBe(1);
});