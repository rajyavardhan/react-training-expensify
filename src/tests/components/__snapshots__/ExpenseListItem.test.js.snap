// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/2"
  >
    <h2>
       
      Rent
       
    </h2>
  </Link>
  <p>
     
    $109,500.00
    -
    Dec 28th 1969
  </p>
</div>
`;
